actionpack/lib/action_dispatch/http/url.rb

```diff
@@ -9,6 +9,7 @@ module URL
       HOST_REGEXP     = /(^[^:]+:\/\/)?(\[[^\]]+\]|[^:]+)(?::(\d+$))?/
       PROTOCOL_REGEXP = /^([^:]+)(:)?(\/\/)?$/
 
+      mattr_accessor :default_protocol, default: "http://"
```

Comment:
`default` is implied here and can be omitted. Think it's odd we're in an HTTP namespace and yet, you could completely rewrite the protocol, can we rephrase to `use_ssl` or `secure_protocol`? And then do `secure_protocol ? "https://" : "http://"